From 338a031e29e4ac514c5bcf0caf4e16ce8afff5df Mon Sep 17 00:00:00 2001
From: Thierry FOURNIER <tfournier@exceliance.fr>
Date: Wed, 26 Feb 2014 18:30:13 +0100
Subject: [PATCH 34/34] BUG/MEDIUM: map: The map parser includes blank lines.

The parser check the end line comparing to the null character.
In fact, the end of line can be also '\r' or '\n'.

The effect is that empty lines are loaded and indexed in maps.

The bug was introduced by commit d5f624dd ("MEDIUM: sample:
add the "map" converter") in 1.5-dev20. No backport is needed.
---
 src/map.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/map.c b/src/map.c
index 28908c5..c3a13a7 100644
--- a/src/map.c
+++ b/src/map.c
@@ -248,7 +248,7 @@ static int map_read_entries_from_file(const char *filename,
 			c++;
 
 		/* empty lines are ignored too */
-		if (*c == '\0')
+		if (*c == '\0' || *c == '\r' || *c == '\n')
 			continue;
 
 		/* look for the end of the key */
-- 
1.8.3.2

